bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,233 @@
+
+    async def lost(self, ctx: commands.Context) -> None:
+        """The player lost the game"""
+        game = self.games[ctx.author.id]
+        game.revealed = game.board
+        await ctx.author.send(":fire: You lost! :fire: ")
+        await game.chat_msg.channel.send(f":fire: {ctx.author.mention} just lost minesweeper :fire:")
+
+    async def won(self, ctx: commands.Context) -> None:
+        """The player won the game"""
+        game = self.games[ctx.author.id]
+        game.revealed = game.board
+        await ctx.author.send(":tada:  You won! :tada: ")
+        await game.chat_msg.channel.send(f":tada: {ctx.author.mention} just won minesweeper :tada:")
+
+    def reveal_zeros(self, revealed: GameDict, board: GameDict, x: int, y: int) -> None:
+        """Recursively reveal adjacent cells when a 0 cell is encountered."""
+        for x_, y_ in self.get_neighbours(x, y):
+            if revealed[y_][x_] != "hidden":
+                continue
+            revealed[y_][x_] = board[y_][x_]
+            if board[y_][x_] == 0:
+                self.reveal_zeros(revealed, board, x_, y_)
+
+    async def check_if_won(self, ctx, revealed: GameDict, board: GameDict) -> bool:
+        """Checks if a player has won"""
+        for x in range(10):
+            for y in range(10):
+                if revealed[y][x] == "hidden" and board[y][x] != "bomb":
+                    return False
+        else:
+            await self.won(ctx)
+            return True
+
+    async def reveal_one(self, ctx: commands.Context, revealed: GameDict, board: GameDict, x: int, y: int) -> bool:
+        """Reveal one square."""
+        revealed[y][x] = board[y][x]
+        if board[y][x] == "bomb":
+            await self.lost(ctx)
+            return True  # game ended
+        elif board[y][x] == 0:
+            self.reveal_zeros(revealed, board, x, y)
+        won = await self.check_if_won(ctx, revealed, board)
+        return won
+
+    @commands.dm_only()
+    @commands.command(name="reveal")
+    async def reveal_command(self, ctx: commands.Context, *Coordinates: CoordinateConverter) -> None:
+        """Reveal multiple cells"""
+        game = self.games[ctx.author.id]
+        revealed: GameDict = game.revealed
+        board: GameDict = game.board
+
+        for x, y in Coordinates:
+            if await self.reveal_one(ctx, revealed, board, x, y):  # game ended
+                await self.update_boards(ctx)
+                del self.games[ctx.author.id]
+                break
+        else:
+            await self.update_boards(ctx)
+
+    @commands.command(name="end")
+    async def end_command(self, ctx: commands.Context):
+        """End the current game"""
+        game = self.games[ctx.author.id]
+        game.revealed = game.board
+        await self.update_boards(ctx)
+        await ctx.author.send(":no_entry: you canceled the game :no_entry:")
+        await game.chat_msg.channel.send(f"{ctx.author.mention} just canceled minesweeper")
```

Comment:
Seems you missed this review the first time:

Perhaps the message with the board could be edited to include this rather than sending a new message? Unlike a win/loss, most people won't care if a game has been cancelled so this message seems a bit spammy.